### PR TITLE
[RDY] Feat/enhance di list

### PR DESCRIPTION
**Pull request by levi85**
Implemented each a  PV invoice views exclusively for each Station Role. The Public role is meant for system accout user. E.g. Finance, Audit and Admin.

### CODE REVIEW

File: src/TestData.js [hunk 1]
```diff
@@ -28,11 +19,11 @@ export const invoicesMock = new Promise(res => {
     rate: '5',
     totalAmountWithGST: Math.floor(Math.random() * 1000),
     tax: 0,
-    status: invoiceStatus[index % 3]
+    status: Object.values(pvStatus)[index % 3]
```
Comment (JingYenLoh) [inline src/TestData.js]: What does this do?

File: src/views/invoice/Invoice.vue [hunk 2]
```diff
@@ -96,20 +101,77 @@ export default {
...
+        case pvStatus.REJECTED:
           return 'Rejected'
+          
         default:
-          return 'Status'
+          return 'Unknown Status'
       }
+    },
+    isCO() {
+      return (this.stationRole == stationRoles.CO && this.pvHeader.status == pvStatus.NEW) ?  true :  false
```
Comment (JingYenLoh) [inline src/views/invoice/Invoice.vue]: Not necessary to use conditional operator, just return the boolean directly

File: src/components/InvoiceTable.vue [hunk 3]
```diff
@@ -29,33 +31,27 @@ export default {
       { text: 'FC Amt', value: 'totalAmountWithGST' },
       { text: 'Status', value: 'status' },
       { text: 'Action', value: 'action' }
-    ]
+    ],
```
Comment (JingYenLoh) [inline src/components/InvoiceTable.vue]: Clean trailing whitespace. I can setup commit hook if necessary

File: src/components/InvoiceTable.vue [hunk 4]
```diff
@@ -65,7 +61,7 @@ export default {
     },
 
     resolveInvoiceRoute(invoiceId) {
-      return `/invoices/${invoiceId}`
+      return this.stationRole != 'default'? `/invoice/${this.stationRole}/${invoiceId}` : `/invoice/${invoiceId}`
```
Comment (JingYenLoh) [inline src/components/InvoiceTable.vue]: Why conditional routing for this?

File: src/TestData.js (hunk 1 above)
Comment (levi85) [reply src/TestData.js]: This is to randomly assign a PV status to the test data, can ignore once we wire up the front-end to backend PVs

File: src/views/invoice/Invoice.vue (hunk 2 above)
Comment (levi85) [reply src/views/invoice/Invoice.vue]: sorry, don't quite understand. The condition operator is setup this way to only display the verify good button on 2 conditions (1) a CO (2) new PV status

Comment (JingYenLoh) [reply src/views/invoice/Invoice.vue]:
```

Comment (levi85) [reply src/views/invoice/Invoice.vue]: oh....got it now. Thanks!

File: src/components/InvoiceTable.vue (hunk 3 above)
Comment (levi85) [reply src/components/InvoiceTable.vue]: what do you mean by commit hook?

File: src/components/InvoiceTable.vue (hunk 4 above)
Comment (levi85) [reply src/components/InvoiceTable.vue]: Beside station officer, non-station officer will also access invoice info. E.g. Finance and Admin so this is to cater for the other roles who will only view the PV without any action required.

**Comment by levi85** [comment]
Changes done.

File: src/components/InvoiceTable.vue (hunk 3 above)
Comment (JingYenLoh) [reply src/components/InvoiceTable.vue]: A commit hook is a custom script that can be fired off when a git commit is made, e.g. pre commit. We can make eslint run pre commit, or configure vue to lint on save which seems to be documented. I'll address this in a separate PR.

**Pull request merged**